ha/core/event_analyzer/parser/parser.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import abc
+import json
+from ha.core.system_health.model.health_event import HealthEvent
+
+class Parser(metaclass=abc.ABCMeta):
+    """
+    Subscriber for event analyzer to pass msg.
+    """
+
+    @abc.abstractmethod
+    def parse_event(self, msg: str) -> HealthEvent:
+        """
+        Parse event.
+        Args:
+            msg (str): Msg
+        """
+        pass
+
+class AlertParser(Parser):
+    """
+    Subscriber for event analyzer to pass msg.
+    """
+
+    def parse_event(self, msg: str) -> HealthEvent:
+        """
+        Parse event.
+        Args:
+            msg (str): Msg
+        """
+        alert = json.loads(msg)
+
+        event = {
+            "event_id" : alert.get('sensor_response_type').get('alert_id'),
```

Comment:
What happen if msg format is one of this field

can we somehow caught and log this exception and raise error

Reply:
Let's add try-catch block for this whole parsing code here?

Reply:
I have removed .get() method and used [] to access the key of the dictionary as .get() method does not raise exception and returns None if the key is not present in the dictionary.
Added try-catch block for the exception handling.

Reply:
ok